gitpod-io/website
# [Analytics] Track initial referrer of session

## Description


Persists the initial referrer of a session in `prevPages` in order to address the issue of the initial referrer not being tracked at the moment

## Related Issue(s)



Fixes [#MKWEB-115](https://linear.app/gitpod/issue/MKWEB-115/[analytics]-first-referrer-of-page-call-sequence-is-not-tracked)

## How to test



1. Open a Preview Environment from this Branch and open preview of website through referrer (easiest way of doing this is clicking one of the links to the preview on the terminal that runs the dev preview
2. Head over to the [Segment Debugger of the Website Staging Source](https://app.segment.com/gitpod/sources/website_staging/debugger) and ensure that the referrer of the `page` call is correct
3. Click on any button on the website that changes the `path` (e.g. to `/cde`) and observe the `page` call once again to check whether the `referrer` continues to be tracked correctly

## Release QA Checklist ✔️



<details>
  <summary>QA Checklist </summary>

## [Priority: High] Functional Testing


## [Priority: High] Content Testing


## [Priority: Medium] Performance Testing


## [Priority: Medium] Accessibility Testing


By including these key items in your website release QA checklist, you can ensure that your website is fully functional, user-friendly, and meets your business requirements.

## Security Testing

> **Note**:💡 If we are changing something related to cookies, data storing, or analytics.


</details>


<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/3868"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
